src/r2mm/profiles/ProfilesClient.ts

```diff
@@ -0,0 +1,33 @@
+import Axios, { AxiosResponse } from 'axios';
+
+export const TSCreateAPI = 'https://thunderstore.io/api/experimental/legacyprofile/create/';
+export const TSRetrieveAPI = (profileImportCode: string) => `https://thunderstore.io/api/experimental/legacyprofile/get/${profileImportCode}/`;
+export const HerokuRetrieveAPI = (profileImportCode: string) => `https://r2modman-hastebin.herokuapp.com/raw/${profileImportCode}`;
+
+export function sendProfile(endpoint: string, payload: string): Promise<AxiosResponse> {
+    return Axios.post(
+        endpoint,
+        payload,
+        {
+            headers: {'Content-Type': 'application/octet-stream'}
+        }
+    );
+}
+
+export async function retrieveProfile(profileImportCode: string): Promise<AxiosResponse> {
+    const instance = Axios.create({
+        validateStatus: function(status) {

```

Comment:
Is this needed?

Reply:
Yes if we don't want to use try catch (mythics suggestion in comments above). Without this Axios will throw an AxiosError and the latter code wont be run. Now that I think about it, if Axios gets something else than a 404 like 500, it probably wont work. I'll change it back to the try catch since it's more reliable.

Reply:
Try catch is fine to use but the point was not everything has to be wrapped in it since you can simply let the promise fail if something unexpected happens.